tool/tctl/common/notification_command.go

```diff
@@ -0,0 +1,308 @@
+	n.rm.Arg("id", "The ID of the notification to remove.").Required().StringVar(&n.title)
+
+	if n.stdout == nil {
+		n.stdout = os.Stdout
+	}
+}
+
+// TryRun takes the CLI command as an argument and executes it.
+func (n *NotificationCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+	nc := client.NotificationServiceClient()
+
+	switch cmd {
+	case n.create.FullCommand():
+		err = n.Create(ctx, client)
+	case n.ls.FullCommand():
+		err = n.List(ctx, nc)
+	case n.rm.FullCommand():
+		err = n.Remove(ctx, nc)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// Create creates a new notification.
+func (n *NotificationCommand) Create(ctx context.Context, client *authclient.Client) error {
+	meta := &headerv1.Metadata{
+		Labels: map[string]string{
+			types.NotificationTitleLabel:       n.title,
+			types.NotificationTextContentLabel: n.content,
+		},
+	}
+
+	subKind := types.NotificationUserCreatedInformationalSubKind
+	if n.warning {
+		subKind = types.NotificationUserCreatedWarningSubKind
+	}
+
+	// Prompt for admin action MFA re-auth.
+	mfaResponse, err := mfa.PerformAdminActionMFACeremony(ctx, client.PerformMFACeremony, true /*allowReuse*/)
+	if err == nil {
+		ctx = mfa.ContextWithMFAResponse(ctx, mfaResponse)
+	} else if !errors.Is(err, &mfa.ErrMFANotRequired) && !errors.Is(err, &mfa.ErrMFANotSupported) {
+		return trace.Wrap(err)
+	}
+
+	nc := client.NotificationServiceClient()
+
+	if n.user != "" {
+		if n.roles != "" || n.requireAllRoles {
+			return trace.BadParameter("roles cannot be configured for a notification which targets a specific user")
+		}
+
+		created, err := nc.CreateUserNotification(ctx, &notificationspb.CreateUserNotificationRequest{
+			Username: n.user,
+			Notification: &notificationspb.Notification{
+				Kind:     types.KindNotification,
+				SubKind:  subKind,
+				Metadata: meta,
+				Spec: &notificationspb.NotificationSpec{
+					Username: n.user,
+				},
+			},
+		})
+
+		if err != nil {
+			return trail.FromGRPC(err)
+		}
+
+		fmt.Printf("\n\ncreated notification: \n%v\n", created)
+
+		fmt.Fprintf(n.stdout, "Created notification %s for user %s\n", created.GetMetadata().GetName(), n.user)
+		return nil
+	}
+
+	if n.roles != "" {
+		created, err := nc.CreateGlobalNotification(ctx, &notificationspb.CreateGlobalNotificationRequest{
+			GlobalNotification: &notificationspb.GlobalNotification{
+				Kind: types.KindGlobalNotification,
+				Spec: &notificationspb.GlobalNotificationSpec{
+					Matcher: &notificationspb.GlobalNotificationSpec_ByRoles{
+						ByRoles: &notificationspb.ByRoles{
+							Roles: n.splitRoles(),
+						},
+					},
+					MatchAllConditions: n.requireAllRoles,
+					Notification: &notificationspb.Notification{
+						Kind:     types.KindNotification,
+						SubKind:  subKind,
+						Metadata: meta,
+						Spec:     &notificationspb.NotificationSpec{},
+					},
+				},
+			},
+		})
+
```

Comment:
```suggestion
```

Keep the error check in the same logical block as the code that generated the error.